### ISSUE TITLE
[FR] Multiple PIN codes

### ISSUE THREAD

**Issue by kalhimeo** [iantrich/restriction-card#67]
**Is your feature request related to a problem? Please describe.**
Not related to a problem, just that at the moment we can only specify one unique pin code per card

**Describe the solution you'd like**
Change the code section to make it a list of strings instead of just a string

**Describe alternatives you've considered**
N.A.

**Additional context**
When multiple users are using the wall tablet, it's nice that that everyone can have his own PIN code (ie. to disable the alarm system).

Thank you very much for considering the feature request :-)

**Comment by user897943** [comment]
I completely agree. Came here to request the same. Multiple occupants in the home, would like each person to have their own code because multiple cards are used to restrict things and I'd like to allow certain codes to use certain cards while blocking on others... If I share my code with the housemate they will be able to unlock other cards I don't want them to have access to.

**Comment by Fusseldieb** [comment]
As a workaround is to create conditional entities based on the logged user. 

Let's say you have three users. Create a card with three alarm entities (the same one three times). Then, using the YAML code editor, conditionally show only the one for the logged user. Do that for all three. Done that, **I think** you can just assign a pin to each one and done.

**Comment by kitty99hub** [comment]
Hi,
@Fusseldieb 
However, this workaround does not necessarily lead to the desired solution!
If, for example, you want to record which person has entered a code on a wall tablet that accesses HA with a separate device login, this is not possible, because in this case, the individual people do not log in to the tablet with their personal login...
The workaround only works with personal devices such as a cell phone, but not on a wall tablet that is permanently logged in with the separate device login.
The desired multi-pin solution would be very helpful here.
I am also looking for a solution on how I can best integrate my Aqara U200 door lock into HA, but for various reasons I have not yet found the right one.
Multiple personal PINs would be great here :)
I will have a look at the pull requests and adding this idea if not already there.
kind regards

**Comment by ildar170975** [comment]
@kalhimeo, some facts: 
1. Currently there is ONE pincode - and it is specified globally.
2. The pincode is "hard-coded" - means "fixed in a card's code", i.e. a user can change it only by changing a yaml code (either inside an UI editor for a storage-mode dashboard OR directly in a yaml file for a yaml-mode dashboard).
3. (for storage-mode dashboard) Changing a card's code in UI editor is not supposed to be done by ANY user - only an admin is supposed to do it (whether it is supported by UI role model or not - is another issue).
4. (for yaml-mode dashboard) ANY user can change yaml files if they have an access & corresponding access rights; how to manage the access - is another issue.
5. If there are several restriction-cards in UI - they are supposed to have same pincode. For storage-mode dashboard it is achieved by simple "copy/paste" - then maintaining same pincode becomes more difficult if you need to change it globally; using a decluttering card helps to keep a pincode in 1 place. For yaml-mode dashboard it is achieved either by yaml-anchor / !include / !secret or decluttering-card.

Assume we need to add multiple pincodes.
There are 2 ways:
1. Pincodes re common for all users - i.e. both John & Jane can use "12345" or "00000".
2. Pincodes are per-user - i.e. John can use "12345" but cannot use "00000" (which can only be used by Jane).

Since we do not have UI editor for `restriction-card` - for both ways we should;
1. Either provide a list of allowed pincodes:
```
code: 
  - 12345
  - 00000
```
2. Or provide a more complex structure:
```
code: 
  - user: aaaa1234567890
    code: 12345
  - user: 1234567890aaaa
    code: 00000
```
where an admin should specify valid user ids.

**Comment by kalhimeo** [comment]
Thank you for coming back on this after all this time 👍

 In my request i was thinking about option 1) , different codes not linked to a specific user (ie: in case of a wall tablet the logged in user doesn't change)

**Comment by ildar170975** [comment]
So, you are requesting to a simple list of codes?
I.e. any user can use any code out of the list?
Could you describe a possible use-case?
The only thought I have is "John like 1234, Jane - 7777, and grandpa can use 0000 for simplicity".

**Comment by kalhimeo** [comment]
Yes that s exactly the use case, each family member uses his own code for convenience. Example: a button to disarm the alarm, protected by pin code. You could also create a new pin code for an external worker without sharing your own  private code.

**Comment by ildar170975** [comment]
@kalhimeo check 1.2.17

**Comment by kalhimeo** [comment]
thank you very much for implementing this !

**Comment by ildar170975** [comment]
You are welcome!
I tested this locally. If you find any issues - please report.

**Comment by Mariusthvdb** [comment]
does this PR now also implement your suggestion to do this:
```
code: 
  - user: aaaa1234567890
    code: 12345
  - user: 1234567890aaaa
    code: 00000
```

or only global multiple pins?

**Comment by ildar170975** [comment]
The simplest scenario is implemented - just a list of codes.

**Comment by ildar170975** [comment]
If anyone wants this:
```
code: 
  - user: aaaa1234567890
    code: 12345
  - user: 1234567890aaaa
    code: 00000
```
please let me know in a separate FR.
